dotnet/src/Connectors/Connectors.Memory.PgVector/PostgresCollection.cs

```diff
@@ -48,9 +48,24 @@ public sealed class PostgresCollection<TKey, TRecord> : VectorStoreCollection<TK
     /// </summary>
     /// <param name="dataSource">The data source to use for connecting to the database.</param>
     /// <param name="name">The name of the collection.</param>
+    /// <param name="ownsDataSource">A value indicating whether the data source should be disposed after the collection is disposed.</param>
     /// <param name="options">Optional configuration options for this class.</param>
-    public PostgresCollection(NpgsqlDataSource dataSource, string name, PostgresCollectionOptions? options = default)
-        : this(new PostgresDbClient(dataSource), name, options)
+    public PostgresCollection(NpgsqlDataSource dataSource, string name, bool ownsDataSource, PostgresCollectionOptions? options = default)
```

Comment:
Don't we want to have a constructor overload that doesn't force the user to specify `ownsDataSource` explicitly?